sentry_core.go

```diff
@@ -193,3 +221,37 @@ func (s *SentryCore) Sync() error {
 	s.hub.Flush(30 * time.Second)
 	return nil
 }
+
+func (s *SentryCore) parseFieldsToEvent(event *sentry.Event, data map[string]interface{}) {
```

Comment:
These functions are not straightforward and simple, I think we need a few tests for them